src/libs/actions/Report.js

```diff
@@ -225,6 +254,14 @@ function getSimplifiedReportObject(report) {
         oldPolicyName,
         visibility,
     };
+
+    const currentMaxSeqNumber = lodashGet(allReports, [report.reportID, 'maxSequenceNumber'], 0);
+
+    // Current local data is newer than report data from fetchReport. For example if Pusher new comment event is comming while API request of fetchReport isn't completed.
+    if (currentMaxSeqNumber > reportMaxSequenceNumber) {
+        return updateSimplifiedReportWithNewerLocalData(simplifiedReport);
+    }
+    return simplifiedReport;
```

Comment:
Ok. I understood your approach but are we sure this is the correct flow of information for the report? I will spend some more time picking pieces on how and when the report is updated to understand that. But you can help me if you have already gone through it.

Reply:
@parasharrajat I think this is correct flow. Every time report is feteched from server it will go to `getSimplifiedReport` to extract necessary data then store it in onyx. 
But in this last commit I am not considering the optimistic action in comparing the maxSequenceNumber here, I have already  made commit in my local to address that. 

Side note: For this particular IOS issue: the only code needed is this part:
```
unreadActionCount: lastReadSequenceNumbers[reportID]
  ? newMaxSequenceNumber - lastReadSequenceNumbers[reportID] - ReportActions.getDeletedCommentsCount(reportID, lastReadSequenceNumbers[reportID])
  : lodashGet(allReports, [reportID, "unreadActionCount"], 0) + 1,
```

Reply:
OK. Thanks for the update. Could you please push your changes so that I can review them all at once? Ready to finish this one.